File: src/main/java/board/Piece.java
package board;

public class Piece {
    public int code;
    public int couleur;
    int nbdir;
    int[] dir;
    boolean glisse;

    public Piece() {
    }

    public Piece(int code, int couleur, int nbdir, int[] dir, boolean glisse) {
        this.code = code;
        this.couleur = couleur;
        this.nbdir = nbdir;
        this.dir = dir;
        this.glisse = glisse;
    }


}

File: src/main/java/board/Piece.kt
package board

class Piece {
    var code: Int = 0
    var couleur: Int = 0
    var nbdir: Int = 0
    lateinit var dir: IntArray
    var glisse: Boolean = false

    constructor()

    constructor(code: Int, couleur: Int, nbdir: Int, dir: IntArray, glisse: Boolean) {
        this.code = code
        this.couleur = couleur
        this.nbdir = nbdir
        this.dir = dir
        this.glisse = glisse
    }
}
